src/server/actions/Organizations.ts

```diff
@@ -19,3 +19,26 @@ export async function softDeleteOrganization(
 
   return res;
 }
+
+/**
+ * update existing organization information
+ * @param organizationId The Id of the organization to be updated
+ * @param updatedData The data to be updated in database
+ * @returns The organization in the database before soft-deletion, or null
+ */
+export async function updateOrganization(
+  organizationId: string,
+  updatedData: object
+): Promise<OrganizationEntity | null> {
+  await dbConnect();
+
+  // Find the organization by its ID and update it with the new data
+  // TODO: should I block updating "softDelete"?
+  const updatedOrganization: OrganizationEntity | null =
+    await OrganizationSchema.findByIdAndUpdate(organizationId, {
+      $set: updatedData,
+      updatedAt: new Date().toJSON(), //set "updatedAt" into timenow
```

Comment:
No need to manually set the `createdAt` or `updatedAt` fields, mongoose does that automatically. This function call can just look like `await findByIdAndUpdate(organizationId, updatedData)`